### DISCUSSION TITLE
Printing Intermediate Steps

### DISCUSSION THREAD

**Discussion by SteveCutler** [buhe/langchain-swift#95] [Ideas]
Is there any way to print intermediate steps of an Agent to console? I see in the agent.swift file there are some references to intermediate steps but all the functionality has been commented out. Was there an issue with it?

**Comment by buhe** [comment]
Such as
```
  let llm = OpenAI(callbacks: [StdOutCallbackHandler()])
  let agent = initialize_agent(llm: llm, tools: [], callbacks: [StdOutCallbackHandler()])
```

**Reply by SteveCutler** [reply]
great thank you!
